### PR TITLE
Option to specify tag for pjax container

### PR DESCRIPTION
**Pull request by alex-code**
Allow the specifying of the container tag, defaults to `<div/>`.
Optional registering of Pjax events against the link and form selectors.

I needed to use Pjax in a table so the option to specify the container as a `<tbody/>` was needed.
I'm also using `$.pjax.reload(...)` so the JavaScript the widget registered wasn't needed.

### CODE REVIEW

File: framework/widgets/Pjax.php [hunk 1]
```diff
@@ -117,7 +121,8 @@ public function init()
     public function run()
     {
         if (!$this->requiresPjax()) {
-            echo Html::endTag('div');
+            $tag = ArrayHelper::remove($this->options, 'tag', 'div');
```
Comment (SilverFire) [inline framework/widgets/Pjax.php]: You remove the `tag` from options array twice: in `run()` and in `init()`

**Comment by SilverFire** [comment]
The feature is good, it may be merged after fixing problem, described in [this comment](https://github.com/yiisoft/yii2/pull/5902#discussion_r46762087)

File: framework/widgets/Pjax.php (hunk 1 above)
Comment (Faryshta) [inline framework/widgets/Pjax.php]: on the previous call the `$this->options` variable is saved on a temporary variable `$options` so `$this->options` is actually unaffected.

Comment (SilverFire) [inline framework/widgets/Pjax.php]: Oh, you're right, sorry

**Comment by SilverFire** [comment]
Merged with minor adjustments, thank you! https://github.com/yiisoft/yii2/commit/e8aa60e41174b94add149d1c533fa19cc17d2bda

**Pull request closed without merging**